.github/workflows/publish.yml

```diff
@@ -0,0 +1,129 @@
+name: Publish image
+
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - '*'
+  repository_dispatch:
+    types: publish
+
+jobs:
+  publish_image:
+    name: Publish image
+    runs-on: ubuntu-latest
+
+    steps:
+    - if: github.event_name == 'repository_dispatch'
+      name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.client_payload.ref }}
+
+    - if: github.event_name != 'repository_dispatch'
+      name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Checkout tags and submodules
+      shell: bash
+      run: |
+        git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+
+        AUTH_HEADER="$(git config --local --get http.https://github.com/.extraheader)"
+        git submodule sync --recursive
+        git -c "http.extraheader=${AUTH_HEADER}" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+
+    - name: Set variables
+      id: vars
+      run: |
+        GITHUB_REPOSITORY_LC=$(echo "${GITHUB_REPOSITORY}" | tr [A-Z] [a-z])
+        echo "::set-output name=name::${GITHUB_REPOSITORY_LC}"
+        echo "Name: ${GITHUB_REPOSITORY_LC}"
+
+        VERSION=$(git describe --tags)
+        echo "::set-output name=version::${VERSION}"
+        echo "Version: ${VERSION}"
+
+        DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+        echo "::set-output name=date::${DATE}"
+        echo "Date: ${DATE}"
+
+        REF=${GITHUB_REF}
+        if [ "${{ github.event_name }}" = "repository_dispatch" ]; then
+          REF="${{ github.event.client_payload.ref }}"
+        fi
+        echo "::set-output name=ref::${REF}"
+        echo "Ref: ${REF}"
+
+        if [ "${REF}" = "refs/heads/master" ]; then
+          ENVIRONMENT="staging"
+          TAG="development"
+          TAGS="dev"
+        else
+          ENVIRONMENT="production"
+          VERSION_MAJOR=$(echo ${VERSION} | sed -r 's/([0-9]+).*/\1/')
+          VERSION_MINOR=$(echo ${VERSION} | sed -r 's/([0-9]+).([0-9]+).*/\1.\2/')
```

Comment:
isn't this both major & minor version numbers? variable's not very clearly named if that's what's intended...